content/en/build/building-with-lotus-APIs.md

```diff
@@ -0,0 +1,646 @@
+
+1. Create a `chainDataQuery.mjs` file in your project and add the following code. We use the `mjs` file format because it supports import statements.
+
+    ```javascript
+    //1. Import necessary packages
+    import { HttpJsonRpcConnector, LotusClient } from "filecoin.js";
+
+    //2. Use the local node URL to create a connection to your Lotus node
+    const localNodeUrl = "http://127.0.0.1:1234/rpc/v0";
+    const localConnector = new HttpJsonRpcConnector({ url: localNodeUrl });
+
+    //3. lotusClient exposes all Lotus APIs
+    const lotusClient = new LotusClient(localConnector);
+    const version = await lotusClient.common.version();
+    console.log(version);
+    ```
+
+2. Run the following command in your project directory:
+
+    ```shell with-output
+    ~/build-with-lotus $ node basicQuery.mjs
+    ```
+
+    ```shell
+    {
+      Version: '1.15.0+calibnet+git.0ac1bbc7a',
+      APIVersion: 66816,
+      BlockDelay: 30
+    }
+    ```
+
+If everything is set up correctly, your code will be able to connect to the Lotus node and return the chain data as shown above. Now, you're ready to explore more API features.
+
+## Query chain data
+
+Blockchain data is a fundamental part of any blockchain network. Using Lotus APIs, you can query chain data, including tipsets, blocks, messages, network status, storage deals, storage provider details, and more.
+
+Let's look at some common chain data-related packages in the filecoin.js library's LotusClient. 
+
+- `LotusClient.common` - methods to get node info
+- `LotusClient.chain` - methods to interact with the blockchain
+- `LotusClient.wallet` - methods to manage wallet on the Lotus node
+- `LotusClient.client` - methods to interact with the storage and retrieval markets as a client
+- `LotusClient.paych` - methods to interact with and manage payment channels
+- `LotusClient.state` - methods to query, inspect, and interact with chain state
+- more ...
+
+In the previous step, we already created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a LotusClient to query chain data. Now let's write more code in there to learn other query functions that do not require auth-token to interact with the Lotus node.
+
+1. Get current chain head
+
+   This function will return the current head of the chain, including current block height, a list of blocks, and a list of CIDs.
+     ```javascript with-output
+      //Query the current block head
+      const chainHead = await lotusClient.chain.getHead();
+      console.log(chainHead);
+     ```
+
+      ```shell
+      {
+        Cids: [
+          {'/': 'bafy2bzacecdkonmhngylnnhrk4azkg2wkgcm6cnm5qn5sk4ww5cszjlvkgkd6'},
+          {'/': 'bafy2bzaceaglcpzhd5gfrzdyt7ce3e5asnbfz3s3stbqyxniziny5snewbpbg'}
+          ],
+        Blocks: [
+          {...},
+          {...}
+          ],
+        Height: 781268
+      }
+      ```
```

Comment:
There's a lot of lines here for a single list item. Should this section be split into smaller headers maybe? Or just remove the `This function will return [...]` bit and tell the user what they need to do. If you need to explain what's going on in the code, do it after you've told them what the code should look like.

Reply:
If I fully post the chain head body here, it will be so many lines (might be 50+ for each block). So I just `...` as an ellipsis for block body. Do you have any suggestions that I can use instead of `...` or it is acceptable?

Reply:
Fixed it.  Thank you, Mosh